docs/adrs/adr-0001-dependency_management.md

```diff
@@ -0,0 +1,53 @@
+- Minimal distruption to developer workflow
+
+## Options
+
+### 1. pip-tools to generate requirements.txt
+Jetstream currently uses this process to manage dependencies.  It involves uses pip-tools to generate a requirements.txt and requirements.in file from the setup.py. [dependabot](https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/) is a built-in github tool that is used to periodically check for dependency updates and create pull requests to implement them.  [Here](https://github.com/mozilla/jetstream/pull/2017) is an example of such a pull request.
+
+* : Dependabot already used in Jetstream
+* +: Jetstream uses it already so there is an existing base of experties in the data team
+* +: Dependencies can be automatically updated with a script like the [update_deps](https://github.com/mozilla/jetstream/blob/main/script/update_deps) script in Jetstream
+* -: Still using pip's dependency resolver, which is fast but can have issues with incompatible versions.  This may be mitigated or even solved by the way pip-tools pins dependencies.
+* -: Jetstream doesn't use `pyproject.toml` for configuring dependencies, though it does seem to be supported by `pip-tools`
+
+### 2. poetry
+(poetry)[https://python-poetry.org/] is a popular tool for dependency managment.  When starting development with a poetry project, one uses the `poetry install` command to create a virtual environment for the project and install the package and its dependencies.  New dependencies are added with the `poetry add` command.  poetry (and other tools) are configured in the `pyproject.toml`.
+
+* : Can add Dependabot, which is alrady built-in to github tool that has additional uses like checking for security vulnerabilities
+* +: Very reliable dependency resolution
+* +: Uses `pyproject.toml`, which is the new standard and is nice for consolidating other config files
+* +: Also acts as a virtualenv manager
+* +: No changes to user workflow
+* +: When a new dependency is added with `poetry add` the `poetry.lock` and `pyproject.toml` files are automatically updated so there minimal risk of a new dependency not being accounted for
+* -: Dependency resolution is slow, when you add a new package it can take several minutes to figure it out
+* -: Developers would have to modify their workflow to accomodate poetry's virtualenv management
+* -: Not wildely use in the data team
+* -: would need to update tox to use poetry to build and publish package
+
+### 3. mamba/conda
+(mamba)[https://mamba.readthedocs.io/en/latest/] is a fast, robust, and cross-platform package manager. It is fully compatible with conda, which is a common package manager in Python, and can be a drop-in replacement for conda. Mamba creates a virtual environment that is isolated from each other.
+
+* Creates isolated virtual environment that encapsulates all the packages in its own designated space
+* Can assign a specific Python version to the given virtual environment
```

Comment:
My biggest concern with conda is that all our docs assume people are using pip to install, and in my experience using pip install in conda envs can lead to big headaches.  Is that still the case